### PR TITLE
cmplxs: replace custom reverse logic with slices.Reverse

### PR DESCRIPTION
**Pull request by jbduncan**
This PR introduces a single modernization change from https://github.com/gonum/gonum/pull/1934. Benchmarks will follow.

Follow-on to https://github.com/gonum/gonum/pull/1934, which was my first attempt at this but was too large to properly benchmark.

Blocked by https://github.com/gonum/gonum/pull/1940.

### CODE REVIEW

**Comment by jbduncan** [comment]
Actually, neither `cmplxs.Reverse` nor its users have any benchmarks, so there's no need to run anything here.

**Comment by jbduncan** [comment]
@kortschak Here are my local benchmark results, running the benchmarks in [dade670](https://github.com/gonum/gonum/pull/1941/commits/dade6701d393f7b9b3f6c9267e851cd9f5f6ce32). (Previously I was using my Linux laptop but now I'm using my M1 MacBook.)

In summary, the difference between `master` and this PR looks to be insignificant.

```
goos: darwin
goarch: arm64
pkg: gonum.org/v1/gonum/cmplxs
                │ benchmarks_cmplxs_reverse_old.txt │ benchmarks_cmplxs_reverse_new.txt  │
                │              sec/op               │   sec/op     vs base               │
ReverseSmall-8                          5.322n ± 0%   5.321n ± 0%       ~ (p=0.576 n=10)
ReverseMedium-8                         397.6n ± 0%   397.9n ± 1%       ~ (p=0.696 n=10)
ReverseLarge-8                          44.26µ ± 1%   43.67µ ± 2%       ~ (p=0.063 n=10)
ReverseHuge-8                           5.335m ± 1%   5.337m ± 1%       ~ (p=0.971 n=10)
geomean                                 4.727µ        4.713µ       -0.30%
```

File: cmplxs/cmplxs.go [hunk 1]
```diff
@@ -564,9 +565,7 @@ func Real(dst []float64, src []complex128) []float64 {
 
 // Reverse reverses the order of elements in the slice.
```
Comment (kortschak) [inline cmplxs/cmplxs.go:566]: Suggest adding deprecation notice pointing to `slices.Reverse`.
```
// Deprecated: This function simply calls [slices.Reverse].
```

Comment (jbduncan) [reply cmplxs/cmplxs.go:567]: Makes sense to me. [Done](https://github.com/gonum/gonum/pull/1941/commits/9ea21b5adf77baec5177d9da143b064cd6c77105).

Comment (jbduncan) [reply cmplxs/cmplxs.go:567]: I'll pore over my other PRs and figure out which ones could do with deprecation notices like this.

**Comment by jbduncan** [comment]
Hi team, this PR is ready, so PTAL.

File: cmplxs/cmplxs_test.go [hunk 2]
```diff
@@ -1371,3 +1371,17 @@ func BenchmarkNorm2Small(b *testing.B)  { benchmarkNorm2(b, Small) }
...
+
+func benchmarkReverse(b *testing.B, size int) {
+	src := rand.NewSource(1)
+	s := randomSlice(size, src)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		Reverse(s)
+	}
+}
+
```
Comment (kortschak) [inline cmplxs/cmplxs_test.go:1383]:
```suggestion
```
for consistency with above.

**Review by kortschak — COMMENTED** [review]
LGTM after nit.

File: cmplxs/cmplxs_test.go (hunk 2 above)
Comment (jbduncan) [reply cmplxs/cmplxs_test.go]: Good catch! Done.

**Comment by jbduncan** [comment]
@kortschak I'm ready for this to be merged in whenever you are.

File: cmplxs/cmplxs.go (hunk 1 above)
Comment (kortschak) [reply cmplxs/cmplxs.go:568]: Make this 
```
// Reverse reverses the order of elements in the slice.
//
// Deprecated: This function simply calls [slices.Reverse].
```

Comment (jbduncan) [reply cmplxs/cmplxs.go:566]: @kortschak Done! :+1:

**Comment by jbduncan** [comment]
@kortschak Ready when you are.

**Review by kortschak — COMMENTED** [review]
Let's merge the last commit into the previous two (the deprecation into the commit that changes the function body and the whitespace change into the commit that adds the bechmarks).

**Comment by jbduncan** [comment]
> Let's merge the last commit into the previous two (the deprecation into the commit that changes the function body and the whitespace change into the commit that adds the bechmarks).

Makes sense! Done. Ready for your next review.

**Review by kortschak — APPROVED** [review]
Thanks

**Comment by jbduncan** [comment]
@kortschak Thank you very much for merging this in! 👍

**Pull request merged**